ocaml/squeezed/src/dune

```diff
@@ -10,7 +10,7 @@
     xapi-stdext-unix
     xapi_version
     astring
-    dune-build-info
```

Comment:
Isn't this needed to generate a correct xapi.version?

Reply:
It is, but it is only needed by xapi_version.ml, everything else only depends on xapi_version.
(transitive dependencies are off now, but that doesn't mean we need to include the transitive deps of xapi_version everywhere: we only need to include them if we use it)

Reply:
The build failures are because dune-build-version is missing, however

Reply:
They were missing in the opam file, fixed: https://github.com/xapi-project/xen-api/pull/5876#discussion_r1688797224